bread/layout/components/datatable.py

```diff
@@ -77,79 +34,63 @@ def from_modelfield(
             else None,
         )
 
+    def as_header_cell(self, orderingurlparameter="ordering"):
+        headcontent = hg.SPAN(self.header, _class="bx--table-header-label")
+        if self.sortingname:
+            headcontent = hg.BUTTON(
+                headcontent,
+                Icon("arrow--down", _class="bx--table-sort__icon", size=16),
+                Icon(
+                    "arrows--vertical",
+                    _class="bx--table-sort__icon-unsorted",
+                    size=16,
+                ),
+                _class=hg.BaseElement(
+                    "bx--table-sort ",
+                    sortingclass_for_column(orderingurlparameter, self.sortingname),
+                ),
+                data_event="sort",
+                title=self.header,
+                **sortinglink_for_column(orderingurlparameter, self.sortingname),
+            )
+        return hg.TH(headcontent, **getattr(self.header, "td_attributes", {}))
+
 
-class DataTable(hg.BaseElement):
+class DataTable(hg.TABLE):
     SPACINGS = ["default", "compact", "short", "tall"]
 
     def __init__(
         self,
         columns: List[DataTableColumn],
-        row_iterator,
-        rowvariable="row",
-        spacing="default",
-        orderingurlparameter="ordering",
-        zebra=False,
+        row_iterator: Union[hg.Lazy, Iterable, hg.Iterator],
```

Comment:
Seing the options of `Iterable` and `hg.Iterator` makes it very clear to me what is going on. I love that.
What still feels a bit like a black box to me are these `hg.Lazy` dudes...

Reply:
As the "java guy" here I would not be against seeing someting like `hg.Lazy[Iterable]` anyway 😄

Reply:
Would ```hg.Lazy[Iterable]``` be comprehensible for a type checker? I think we would need to define it as container-type or something...